### DISCUSSION TITLE
File modified time not updating under windows - expected behaviour?

### DISCUSSION THREAD

**Discussion by neilstephens** [gabime/spdlog#2644] [Q&A] [answered]
Gday,

I'm sorry if this has been pointed out before (I expect it has, but I can't find any previous issues with my searches).

I'm using rotating_file_sink, and everything is working as expected. Logs are regularly flushing to disk, and I can verify by reloading the file in an editor etc.

Interestingly, the file modified time in windows doesn't update until the file rotates, even though regular writes are happening.
Under linux, the modified time updates after (what seems like) every write.

I understand this might be expected behaviour, but I'd like to know why this happens? I got to the point of seeing that there's  fsopen and std::fwrite under the hood, and MS doco that I found for those functions doesn't mention anything about timestamps etc.

The reason for stumbling accross this is that "follow" mode in notepad++ wasn't working (similar to how I might tail a log under linux), so I guess someone else would have raised this before?

Cheers.

**Comment by tt4g** [comment, marked as answer]
spdlog does not explicitly call the API to change the Last Modified Date.
This is due to the behavior of the Windows OS and file system.

Stack Overflow has [a very similar question](https://stackoverflow.com/questions/51611819/file-last-modified-not-updating-when-java-writes-to-windows-server-2016) and [a great answer](https://stackoverflow.com/a/51617747) on when Last Modified Date changes in Windows OS.

Quote answer (https://stackoverflow.com/a/51617747):

> [Flush is not the same as sync](https://stackoverflow.com/questions/4072878/i-o-concept-flush-vs-sync). The FileHandler just performs a flush after each record is published. Windows is not setup to force a writing of the metadata to the file system. From [File “Date modified” property are not updating while modifying a file without closing it.](https://blogs.technet.microsoft.com/asiasupp/2010/12/14/file-date-modified-property-are-not-updating-while-modifying-a-file-without-closing-it/):
